### PR TITLE
updated dark mode

### PR DESCRIPTION
**Pull request by alexanderalkhazov**
updated dark mode with use context what do you think ?

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,41 +1,62 @@
...
 import More from "./pages/More/More";
 import Forecast from "./pages/Forecast/Forecast";
+import { ThemeProvider } from "./providers/ThemeContext";
+import "./App.css";
 
-const App = () => {
-  const [darkMode, setDarkMode] = useState(false);
+const App: React.FC = () => {
   const [units, setUnits] = useState(false);
+  const [coords, setCoords] = useState([0, 0]); // array of 2 items of type number to represent lat and lon for geolocation
```
[the remark this answers is not in the document]
Comment (DaniilGL) [reply src/App.tsx:12]: will be more readable

File: src/App.tsx [hunk 2]
```diff
@@ -1,41 +1,62 @@
...
 
-  const toggleDarkMode = () => {
-    setDarkMode(!darkMode);
-  };
+  const [data, setData] = useState({
```
Comment (DaniilGL) [inline src/App.tsx:14]: create a const for the defaultData

File: src/App.tsx [hunk 3]
```diff
@@ -1,41 +1,62 @@
...
+    temp: 60,
+    humidity: "20%",
+    windSpeed: "20 kmh",
+    description: "good weather very nice",
+  });
 
   useEffect(() => {
-    if (darkMode) {
-      document.body.classList.add("dark-mode");
+    if (navigator.geolocation) {
```
Comment (DaniilGL) [inline src/App.tsx:23]:
```suggestion
    if (!navigator.geolocation) {
   console.log("Geolocation is not supported by this browser.")
    return
    }
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,41 +1,62 @@
...
+      navigator.geolocation.getCurrentPosition(function (position) {
+        const latitude = position.coords.latitude;
+        const longitude = position.coords.longitude;
+        console.log(`Latitude: ${latitude}, Longitude: ${longitude}`);
```
Comment (DaniilGL) [inline src/App.tsx:27]: remove logs

File: src/providers/ThemeContext.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { createContext, useState, useEffect, useContext } from 'react';
+
+interface ThemeContextProps {
+  darkMode: boolean;
```
Comment (DaniilGL) [inline src/providers/ThemeContext.tsx:4]: why darkMode?
call its currentTheme
and then add an enum
enum Themes {light,dark}
or something not sure about the name

File: src/providers/ThemeContext.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  useEffect(() => {
+    const storedTheme = localStorage.getItem('darkMode');
+    if (storedTheme) {
+      setDarkMode(storedTheme === 'true');
+    }
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem('darkMode', darkMode.toString());
+    document.body.classList.toggle('dark-mode', darkMode);
```
Comment (DaniilGL) [inline src/providers/ThemeContext.tsx:30]: [you ](https://blog.logrocket.com/create-better-themes-with-css-variables/)](https://blog.logrocket.com/create-better-themes-with-css-variables/)

File: src/providers/ThemeContext.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+export const ThemeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const [darkMode, setDarkMode] = useState(false);
+
+  useEffect(() => {
+    const storedTheme = localStorage.getItem('darkMode');
+    if (storedTheme) {
+      setDarkMode(storedTheme === 'true');
+    }
```
Comment (DaniilGL) [inline src/providers/ThemeContext.tsx:25]:
```suggestion
    const storedTheme = localStorage.getItem('darkMode');
    if (storedTheme) {
      setTheme(storedTheme||DEFAULT_THEME);
    }
```

File: src/providers/ThemeContext.tsx (hunk 6 above)
Comment (DaniilGL) [inline src/providers/ThemeContext.tsx:30]: document body?

File: src/pages/Forecast/Forecast.tsx [hunk 8]
```diff
@@ -1,6 +1,9 @@
 
+interface IForecastProps {
+  data: any;
```
Comment (DaniilGL) [inline src/pages/Forecast/Forecast.tsx:3]: any

File: src/pages/CurrentLocation/CurrentLocation.tsx [hunk 9]
```diff
@@ -3,22 +3,23 @@ import "./CurrentLocation.css";
 interface ICurrentLocationProps {
   units: boolean;
   toggleUnits: any;
+  data: any;
```
Comment (DaniilGL) [inline src/pages/CurrentLocation/CurrentLocation.tsx:6]: any

File: src/pages/CurrentLocation/CurrentLocation.tsx [hunk 10]
```diff
@@ -3,22 +3,23 @@ import "./CurrentLocation.css";
 interface ICurrentLocationProps {
```
Comment (DaniilGL) [inline src/pages/CurrentLocation/CurrentLocation.tsx:3]: I prefer just calling it Props youre not sharing it with different components

File: src/App.tsx [hunk 11]
```diff
@@ -1,41 +1,65 @@
...
+    description: "good weather very nice",
+  });
 
   useEffect(() => {
-    if (darkMode) {
-      document.body.classList.add("dark-mode");
+    if (!navigator.geolocation) {
+   console.log("Geolocation is not supported by this browser.")
+    return
+    }
```
Comment (DaniilGL) [inline src/App.tsx:26]: space after the if

File: src/App.tsx [hunk 12]
```diff
@@ -1,41 +1,65 @@
...
+      navigator.geolocation.getCurrentPosition(function (position) {
```
Comment (DaniilGL) [inline src/App.tsx:27]: use arrow function

File: src/App.tsx [hunk 13]
```diff
@@ -1,41 +1,65 @@
...
+        const latitude = position.coords.latitude;
+        const longitude = position.coords.longitude;
+        console.log(`Latitude: ${latitude}, Longitude: ${longitude}`);
+        setCoords([latitude, longitude]);
```
Comment (DaniilGL) [inline src/App.tsx:31]: fix it to object

File: src/Components/DarkModeSwitch/DarkModeSwitch.tsx [hunk 14]
```diff
@@ -1,20 +1,19 @@
...
   return (
-    <>
-      <label
-        className={`switch ${darkMode ? "dark" : "light"}`}
-        onClick={toggleDarkMode}
-      >
-        <span className="slider"></span>
-      </label>
-    </>
+    <label className={`switch ${darkMode ? 'dark' : 'light'}`} onClick={toggleTheme}>
```
Comment (DaniilGL) [inline src/Components/DarkModeSwitch/DarkModeSwitch.tsx:14]: omg create a const for it

File: src/App.tsx (hunk 2 above)
Comment (alexanderalkhazov) [reply src/App.tsx:14]: const deafulatData = {
    currentLocation: "Petah tiqva",
    temp: 60,
    humidity: "20%",
    windSpeed: "20 kmh",
    description: "good weather very nice",
  };
  
  const [data, setData] = useState(deafulatData);

File: src/App.tsx (hunk 4 above)
Comment (alexanderalkhazov) [reply src/App.tsx:30]: done

File: src/Components/DarkModeSwitch/DarkModeSwitch.tsx (hunk 14 above)
Comment (alexanderalkhazov) [reply src/Components/DarkModeSwitch/DarkModeSwitch.tsx:14]: what do you mean please

File: src/pages/Forecast/Forecast.tsx (hunk 8 above)
Comment (alexanderalkhazov) [reply src/pages/Forecast/Forecast.tsx:3]: what to type ? right now its working with fake data i dont have interfaces yet

**Pull request closed without merging**